llvm/test/Transforms/LoopVectorize/AArch64/interleave_count.ll

```diff
@@ -0,0 +1,106 @@
+; RUN: opt < %s -force-vector-width=64 -O3 -S -pass-remarks=loop-vectorize 2>&1 | FileCheck %s
```

Comment:
please don't use `-O3`, as this makes the test prone to failing due to changes in other parts of the pipeline.

Also, what's the motivation for forcing vector-width to 64 here? It is an impossible VF for AArch64. Would it be possible to just let the cost model also pick the VF?

Reply:
> Also, what's the motivation for forcing vector-width to 64 here? It is an impossible VF for AArch64. Would it be possible to just let the cost model also pick the VF?

The purpose of the test is to show the impact of interleaving count computation change that will be added in a subsequent patch https://github.com/llvm/llvm-project/pull/70141. The IC is a function of VF & TC (trip count) and to test this change the VF needs to be large enough that the vectorized-interleaved loop runs once or twice. However, the TC threshold to allow interleaving is 128, so I increased the VF to hit this case. 

On trying to work around this by reducing the TC & the TC threshold with `-tiny-trip-count-interleave-threshold` option, the loop is not getting vectorized because of hitting a clause of runtime pointer checks.

Reply:
Ah I see. The runtime check issue should be resolvable by `define void @loop_with_tc_128(ptr noalias %p, ptr noalias %q) {`

Reply:
I reduced the VF to 8 & made relevant changes to show the same intended impact. 
I forced a VF because otherwise the vectorization threshold will come into play, adding one more configuration option. For a VF smaller than 8, I'll have to tune other parameters like small-loop-cost to make these tests show the IC computation without getting blocked earlier.

Reply:
IIUC this test should ideally check the cost model selecting the expected VF and IC. Forcing a VF seems to leave a gap in testing here and it would be better to not artificially force the VF, in particular because the cos model is expected to pick 16 for the loops?

Reply:
Increased the loop trip count to hit the intended case of changed interleave count computation in subsequent patches without forcing VF.